ui/components/app/simulation-details/simulation-details.tsx

```diff
@@ -147,43 +122,50 @@ const SimulationPreviewLayout: React.FC = ({ children }) => (
  * @param props
  * @param props.simulationData - The simulation data to display.
  */
-export const SimulationPreview: React.FC<SimulationPreviewProps> = ({
+export const SimulationDetails: React.FC<SimulationPreviewProps> = ({
   simulationData,
 }: SimulationPreviewProps) => {
   const t = useI18nContext();
+  const { pending: loading, value: balanceChanges } =
+    useBalanceChanges(simulationData);
+
+  if (loading) {
+    return (
+      <SimulationPreviewLayout
+        inHeader={<LoadingIndicator />}
+      ></SimulationPreviewLayout>
+    );
+  }
 
-  const simulationFailed = !simulationData;
-  if (simulationFailed) {
+  const error = !simulationData;
```

Comment:
The error state is now explicit via the presence of an `error` property in the `simulationData`.

It includes `code`, `message`, and `isReverted` properties.

`undefined` now means we're waiting for the simulation response so that also needs to be reflected in the `loading` boolean and pre-loader visibility.

As for the chain not being supported, we haven't yet added an explicit property so we can check if `error.message` includes `Chain is not supported`.